internal/pkg/service/appsproxy/dataapps/api/config.go

```diff
@@ -10,61 +11,83 @@ import (
 	"github.com/keboola/keboola-as-code/internal/pkg/service/appsproxy/dataapps/auth/provider"
 )
 
+// maxCacheExpiration is the maximum duration for which an old AppConfig of a data app is cached.
+const maxCacheExpiration = time.Hour
+
 type AppID string
 
-type AppProxyConfig struct {
+type AppConfig struct {
 	ID             AppID              `json:"-"`
 	Name           string             `json:"name"`
 	UpstreamAppURL string             `json:"upstreamAppUrl"`
 	AuthProviders  provider.Providers `json:"authProviders"`
 	AuthRules      []Rule             `json:"authRules"`
 	eTag           string
-	modified       bool
 	maxAge         time.Duration
 }
 
+type NotModifiedError struct {
+	MaxAge time.Duration
+}
+
+func (v NotModifiedError) Error() string {
+	return "app proxy config: not modified"
+}
+
 func (c AppID) String() string {
 	return string(c)
 }
 
-func (c *AppProxyConfig) IsModified() bool {
-	return c.modified
+func (c AppConfig) ETag() string {
+	return c.eTag
 }
 
-// GetAppProxyConfig loads proxy configuration for the specified app.
-// eTag is used to detect modifications, if the eTag doesn't match, the AppProxyConfig.IsModified method returns true.
-func (a *API) GetAppProxyConfig(appID AppID, eTag string) request.APIRequest[*AppProxyConfig] {
-	result := &AppProxyConfig{}
+func (c AppConfig) MaxAge() time.Duration {
+	return c.maxAge
+}
+
+// GetAppConfig loads proxy configuration for the specified app.
+// eTag is used to detect modifications, if the eTag match, the NotModifiedError is returned.
+func (a *API) GetAppConfig(appID AppID, eTag string) request.APIRequest[*AppConfig] {
+	result := &AppConfig{}
 	return request.NewAPIRequest(result, a.newRequest().
 		WithResult(result).
 		WithGet("apps/{appId}/proxy-config").
 		AndPathParam("appId", appID.String()).
 		AndHeader("If-None-Match", eTag).
 		WithOnSuccess(func(ctx context.Context, response request.HTTPResponse) error {
-			// Add app id to the result
-			result.ID = appID
-
-			// Add ETag to result
-			result.eTag = response.ResponseHeader().Get("ETag")
-
 			// Process Cache-Control header
-			cacheControl := response.ResponseHeader().Get("Cache-Control")
-			if cacheControl == "" {
-				return nil
+			var maxAge time.Duration
+			if cacheControl := response.ResponseHeader().Get("Cache-Control"); cacheControl != "" {
+				if cacheDirectives, err := cacheobject.ParseResponseCacheControl(cacheControl); err != nil {
+					return err
+				} else if cacheDirectives.MaxAge > 0 {
+					maxAge = minDuration(maxCacheExpiration, time.Second*time.Duration(cacheDirectives.MaxAge))
+				}
 			}
 
-			cacheDirectives, err := cacheobject.ParseResponseCacheControl(cacheControl)
-			if err != nil {
-				return err
+			// Return specific error if there is no content, because StatusNotModified
+			if response.StatusCode() == http.StatusNotModified {
+				return NotModifiedError{MaxAge: maxAge}
 			}
```

Comment:
If the API returns `304 - NotModified`, there is no content returned :exclamation: 
So it must be error, otherwise, it leads to difficult-to-debug problems when an empty value appears somewhere.